notebooks/Create Asset and Events.ipynb

```diff
@@ -0,0 +1,192 @@
+    "\n",
+    "filterwarnings(\"ignore\", message=\"Unverified HTTPS request\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "05c9e0bc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Connection parameters.  Fill in with client ID and secret from an appropriately \n",
+    "# permissioned Application Registration\n",
+    "RKVST_URL=\"https://app.rkvst.io\"\n",
+    "# APPREG_CLIENT=\"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx\"\n",
+    "# APPREG_SECRET=\"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b3bff987",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_artist(arch, name, artist_name, genre, artistid):\n",
+    "    \"\"\"\n",
+    "    Creates an asset that passes name, stage name, music genre and unique id\n",
+    "    for an Artist\n",
+    "    \"\"\"\n",
+    "    attrs = {\n",
+    "        \"arc_display_name\": name,\n",
+    "        \"arc_description\": \"British Soul Singer\",\n",
+    "        \"arc_display_type\": \"Artists\",\n",
+    "        \"stage_name\": artist_name,\n",
+    "        \"genre\": genre,\n",
+    "        \"artistid\": artistid\n",
+    "    }\n",
+    "    \n",
+    "    return arch.assets.create(attrs=attrs, confirm=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5a8456d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_release(arch, asset, album_name, release):\n",
+    "    \"\"\"\n",
+    "    Creates an event that passes album name and release year\n",
+    "    \"\"\"\n",
+    "    props = {\n",
+    "        \"operation\": \"Record\",\n",
+    "        \"behaviour\": \"RecordEvidence\"\n",
+    "    }\n",
+    "    attrs = {\n",
+    "        \"arc_description\": \"Artist Information\",\n",
+    "        \"arc_display_type\": \"Album Release\",\n",
+    "        \"album_name\": album_name,\n",
+    "        \"relase_year\": release\n",
+    "    }\n",
+    "    \n",
+    "    return arch.events.create(asset[\"identity\"], props=props, attrs=attrs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d795048b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "Main function of asset and event creation.\n",
+    "\n",
+    "* Connect to RKVST with client ID and client secret\n",
+    "* Creates an Asset and two events\n",
+    "* Prints response of asset and event creation\n",
+    "\"\"\"\n",
+    "# Optional call to set the logger level.  The argument can be either \n",
+    "# \"INFO\" or \"DEBUG\".  For more sophisticated logging control see our \n",
+    "# documentation.\n",
+    "set_logger(\"INFO\")\n",
+    "\n",
+    "# Initialize connection to RKVST\n",
+    "print(\"Connecting to RKVST\")\n",
+    "arch = Archivist(RKVST_URL, (APPREG_CLIENT, APPREG_SECRET), max_time=300)"
```

Comment:
Hey @eccles   . . . using "with" changes the cells in a way where all the logic for creating assets and events is held in one cell, thus removing the capability to step through the execution of creating an asset and then creating the event . . . it happens on one cell.

Wondering does this subtracts from "dead simple" idea/concept, do we think this will "add" confusion or as long as we have simple but relevant comments we should be ok?

Thoughts?

cc: @jag

Reply:
for what it is worth, I think 'flat is better than nested' wins here. we are not trying to teach good python programming. A note or a comment somewhere that sais "the rkvst python sdk is authored to work cleanly with context managers. our examples are written for clarity rather than necessarily best practice" might be better ?